src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -88,27 +87,28 @@ def __init__(self, *args, **kwargs):
     def validate(self, data):
         data['arrayjoin'] = self.arrayjoin
 
-        return data
-
-    def validate_range(self, attrs, source):
-        has_start = bool(attrs.get('start'))
-        has_end = bool(attrs.get('end'))
-        has_range = bool(attrs.get('range'))
-
-        if has_start != has_end or has_range == has_start:
+        if data.get('start') or data.get('end'):
+            if not all([data.get('start'), data.get('end')]):
+                raise InvalidParams('start and end are both required')
+            return data
+
+        # Populate start and end if they weren't directly provided
+        try:
+            start, end = get_date_range_from_params({
+                'statsPeriod': data.get('statsPeriod') or data.get('range'),
+                'statsPeriodStart': data.get('statsPeriodStart'),
+                'statsPeriodEnd': data.get('statsPeriodEnd'),
```

Comment:
Are these new params intended to replace the current `start`, `end`, `statsPeriod` parameters the front end is using? The global search header component has some strong opinions on what the query string parameters need to be as it aims to keep them consistent across all the endpoints.

Reply:
no it's not replacing anything. it's adding additional functionality -- the ability to specify a range like "14 days ago until 7 days ago" which Billy needs for some dashboard functionality (specifically, showing data from a previous time window for comparison)